### PR TITLE
Implement testing with Jest and React-testing-library

### LINKED ISSUE #26: Setting up infrastructure for Testing
- Jest & React testing library are the main tools 
  - RTL install docs can be found here: https://testing-library.com/docs/react-testing-library/intro

Setting up codebase for testing can be found in Gatsby docs here:  https://www.gatsbyjs.org/docs/unit-testing/

### PR DESCRIPTION
**Pull request by benfluleck**
*Description*
This PR implements React-testing-library and Jest
- Create config for tests
- Install setup config

*Screenshots*
Screenshot 2019-10-14 at 6 39 25 PM

closes #26

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -11,16 +11,30 @@
...
+      "<rootDir>/src/config/setupTests.js"
+    ]
+  },
   "repository": {
     "type": "git",
     "url": "https://github.com/JacobMGEvans/lake-county-tech-meetup",
     "website": "https://lake-county-tech-meetup.netlify.com"
   },
   "dependencies": {
+    "@testing-library/jest-dom": "^4.1.2",
```
Comment (JacobMGEvans) [inline package.json]: Was this required by the Gatsby Docs or you thought it was a good addition?

File: src/__tests__/components/Footer.test.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  }
+
+}
+
+describe('Footer Component', () => {
+
+  test('should render', () => {
+    const { container } = setup()
+
+    expect(container).toBeTruthy()
```
Comment (JacobMGEvans) [inline src/__tests__/components/Footer.test.js]: I won't block for this, but for future reference, I prefer practical &  sound tests. Testing if the React DOM outputted "something" that wasn't null, undefined, or false... It is too general. 

Any blog by Kent C. Dodds is great for tests here is one: https://kentcdodds.com/blog/making-your-ui-tests-resilient-to-change

File: src/config/setupTests.js [hunk 3]
```diff
@@ -0,0 +1,6 @@
+import React, { Fragment, Component } from 'react'
+import "@testing-library/jest-dom/extend-expect"
+
+global.React = React
+global.Fragment = Fragment
+global.Component = Component
```
Comment (JacobMGEvans) [inline src/config/setupTests.js]: Is this file also from the Gatsby instructions for setting up tests?

**Comment by JacobMGEvans** [comment]
Love that you added the screenshot!

**Comment by JacobMGEvans** [comment]
closes #26

File: package.json (hunk 1 above)
Comment (benfluleck) [reply package.json]: This package helps with custom jest matchers like `toHaveAttribute, toBeDisabled, toBeValid` e.t.c

https://github.com/testing-library/jest-dom

File: src/config/setupTests.js (hunk 3 above)
Comment (benfluleck) [reply src/config/setupTests.js]: This is the config file that runs after the tests. It is advised in the Gatsby docs https://www.gatsbyjs.org/docs/testing-react-components/

We use this in our codebase as well.

line 4,6 is to help scope React and stop us importing it in the tests.

File: src/__tests__/components/Footer.test.js (hunk 2 above)
Comment (benfluleck) [reply src/__tests__/components/Footer.test.js]: Truth is I didn't want to anything too specific as I didn't think it was under the scope 😄

File: src/config/setupTests.js (hunk 3 above)
Comment (benfluleck) [reply src/config/setupTests.js]: @JacobMGEvans  feedback implemented

**Comment by benfluleck** [comment]
@JacobMGEvans  feedback implemented

**Comment by JacobMGEvans** [comment]
Congrats!

**Pull request merged**